### PR TITLE
Add extra possibility for username and emoij to make it more Tekton Like

### PR DESCRIPTION
**Pull request by GijsvanDulmen**
# Changes

Add a few more parameters to Slack webhook to make it more Tekton like in a Slack channel.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [X] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [X] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [X] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [X] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [X] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [X] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [X] mandatory `spec.description` follows the convention

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/send-to-webhook-slack/0.1/send-to-webhook-slack.yaml [hunk 1]
```diff
@@ -20,15 +21,32 @@ spec:
...
   steps:
   - name: post
     image: docker.io/curlimages/curl:7.68.0@sha256:99a8e9629b3ae26efb977e1a98f4786d6bd730c5ab4dea64632e297d7c3e7151 #tag: 7.68.0
     script: |
-      #!/bin/sh
-      /usr/bin/curl -X POST -H 'Content-type: application/json' --data '{"text":"$(params.message)"}' $URL
+      #!/usr/bin/env sh
+      MESSAGE=$(echo "${MESSAGE}" | sed -e 's/\"/\\\\"/g')
+      BOTNAME=$(echo "${BOTNAME}" | sed -e 's/\"/\\\\"/g')
+      JSON="{\"text\": \"${MESSAGE}\", \"username\": \"${BOTNAME}\", \"icon_emoji\": \"${EMOJI}\"}"
```
Comment (chmouel) [inline task/send-to-webhook-slack/0.1/send-to-webhook-slack.yaml:39]: what happend if we don't specify those EMOJI and BOTNAME would it fails?

File: task/send-to-webhook-slack/0.1/send-to-webhook-slack.yaml [hunk 2]
```diff
@@ -20,15 +21,32 @@ spec:
   - name: message
     type: string
     description: plain text message
+  - name: bot-name
+    type: string
+    description: plain text message
+    default: 'Tekton Bot'
```
Comment (chmouel) [inline task/send-to-webhook-slack/0.1/send-to-webhook-slack.yaml:27]: is there a way to make it conditional if the user doesn't want it?

Comment (GijsvanDulmen) [reply task/send-to-webhook-slack/0.1/send-to-webhook-slack.yaml:27]: Yes there is. If you don't send a username the name of the bot becomes "Incoming webhook". See attached screenshot as an example. (The icon_emoji in our case was ":tekton:" which was an imported emoji. I think adding a username is better than providing none.

Schermafbeelding 2021-06-02 om 19 31 58


Schermafbeelding 2021-06-02 om 19 32 02

File: task/send-to-webhook-slack/0.1/send-to-webhook-slack.yaml [hunk 3]
```diff
@@ -20,15 +21,32 @@ spec:
...
+  - name: icon-emoji
+    type: string
```
Comment (vinamra28) [inline task/send-to-webhook-slack/0.1/send-to-webhook-slack.yaml:29]: can we please update the README as well with these new params?

**Review by vinamra28 — COMMENTED** [review]
/ok-to-test

File: task/send-to-webhook-slack/0.1/send-to-webhook-slack.yaml (hunk 3 above)
Comment (GijsvanDulmen) [reply task/send-to-webhook-slack/0.1/send-to-webhook-slack.yaml:29]: Added docs in the README. Forget them :-(

**Comment by chmouel** [comment]
/lgtm

Looking Good! 🤙🏽

**Comment by vinamra28** [comment]
one last thing :sweat_smile: can you please squash the commits?

**Comment by GijsvanDulmen** [comment]
> one last thing 😅 can you please squash the commits?

Done! Can be merged now!

**Review by vinamra28 — APPROVED** [review]
/lgtm
Thanks for the patch

**Pull request merged**